system_metrics_collector/src/topic_statistics_collector/subscriber_topic_statistics.hpp

```diff
@@ -80,13 +80,14 @@ class SubscriberTopicStatisticsNode : public system_metrics_collector::MetricsMe
 
     const auto collect_topic_desc = BuildTopicParameterDescriptor(
       "The topic to subscribe to, for calculating topic statistics.");
-    collect_topic_name_ = declare_parameter(
+    collect_topic_names_ = declare_parameter(
       constants::kCollectStatsTopicNameParam,
-      std::string() /* setting default to empty, since this is a required parameter */,
+      std::vector<std::string>() /* setting default to empty, since this is a required parameter */,
       collect_topic_desc);
-    ValidateStringParam(
-      constants::kCollectStatsTopicNameParam,
-      collect_topic_name_);
+
+    if (collect_topic_names_.empty()) {
+      ValidateStringParam(constants::kCollectStatsTopicNameParam, std::string());
```

Comment:
This check is redundant - if the input vector is empty, this is checking a non-existent string by passing an empty string to the validate function. I think we just want to have the desired behaviour in case of empty params here instead of a call to `ValidateStringParam()`.